lib/logstash/plugin_mixins/http_client.rb

```diff
@@ -19,16 +19,22 @@ def self.included(base)
   end
 
   class Adapter < Module
-    def initialize(with_deprecated: false)
+    def initialize(with_deprecated: false, with_obsolete: false)
+       raise ArgumentError, "A plugin cannot support deprecated and obsolete SSL settings" if with_deprecated && with_obsolete
```

Comment:
I see how *today* this is the case, but it seems like there is a world in which over a plugin's lifetime we would want to have both deprecated and obsolete config?

Reply:
I was thinking that there was a case where we come up with a *new* deprecation we want to share across modules. In this case we would still want to alert on the last batch or obsolete settings in addition to the new deprecated one. Maybe that is not a valid scenario though. 

Seems like instead of having a mutually exclusive set of obsolete/deprecated options consumers would want a specific version. But that would be a pretty significant refactor. I suppose if the use case comes up (a shared new deprecated parameter) we can always just add it to the `ObsoleteSslConfigSupport` implementation so we are not at risk of boxing ourselves in with this.

Reply:
I did think about doing that way, but my fear of doing it as a specific version of the plugin was the danger of inadvertently including the wrong version of the plugin as a dependency - being explicit about the obsolescence reduces the chance of a misconfigured gemspec blowing up a plugin